boring/src/bio.rs

```diff
@@ -23,12 +23,7 @@ impl<'a> MemBioSlice<'a> {
         ffi::init();
 
         assert!(buf.len() <= c_int::max_value() as usize);
-        let bio = unsafe {
-            cvt_p(BIO_new_mem_buf(
-                buf.as_ptr() as *const _,
-                buf.len() as c_int,
-            ))?
-        };
```

Comment:
These signatures use `c_int` in the pinned FIPS revision and `usize` in the new revision. While Signal doesn't use the FIPS module, we don't want to diverge from upstream cloudflare/boring in an incompatible way. This `as _` idiom is seen in a few other places in the project, most notably calls to `OPENSSL_malloc`.